windows/src/engine/keyman32-ver/calldll.cpp

```diff
@@ -227,6 +227,16 @@ BOOL UnloadDLLs(LPINTKEYBOARDINFO lpkbi)
 
   if (lpkbi->lpCoreKeyboardState) {
           km_kbp_state_imx_deregister_callback(lpkbi->lpCoreKeyboardState);
+          // Need to Load the keyman32 or keyman64 dll proxy keyboard as the third-party dlls
+          // will use this rather than the versioned dll.
+          #ifdef _WIN64
+            HMODULE hModule = GetModuleHandle("keyman64.dll");
+          #else
+            HMODULE hModule = GetModuleHandle("keyman32.dll");
+          #endif
+          if (hModule){
+            FreeLibrary(hModule);
+          }
```

Comment:
It's better to be caching module handle per-thread, so store it in the global thread data I think. Otherwise we run the risk of race conditions in multi-threaded apps where the dll may be freed out under itself.